tests/components/inels/conftest.py

```diff
@@ -0,0 +1,125 @@
+"""Test fixtures."""
+
+from typing import Any
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from homeassistant.core import HomeAssistant
+
+from . import HA_INELS_PATH
+from .common import DOMAIN, MockConfigEntry, get_entity, set_mock_mqtt
+
+
+@pytest.fixture(autouse=True)
+def auto_enable_custom_integrations(enable_custom_integrations: None):
+    """Enable custom integrations for testing."""
+    return
+
+
+@pytest.fixture(name="mock_mqtt")
+def mock_inelsmqtt_fixture():
+    """Mock inels mqtt lib."""
+
+    def messages():
+        """Return mocked messages."""
+        return mqtt.mock_messages
+
+    def last_value(topic):
+        """Mock last_value to return None if mock_last_value is empty or topic doesn't exist."""
+        return mqtt.mock_last_value.get(topic) if mqtt.mock_last_value else None
+
+    async def discovery_all():
+        """Return mocked discovered devices."""
+        return mqtt.mock_discovery_all
+
+    async def subscribe(topic, qos=0, options=None, properties=None):
+        """Mock subscribe fnc."""
+        if isinstance(topic, list):
+            return {t: mqtt.mock_messages.get(t) for t in topic}
+        return mqtt.mock_messages.get(topic)
+
+    async def publish(topic, payload, qos=0, retain=True, properties=None):
+        """Mock publish to change value of the device."""
+        mqtt.mock_messages[topic] = payload
+
+    unsubscribe_topics = AsyncMock()
+    unsubscribe_listeners = Mock()
+
+    mqtt = Mock(
+        messages=messages,
+        subscribe=subscribe,
+        publish=publish,
+        last_value=last_value,
+        discovery_all=discovery_all,
+        unsubscribe_topics=unsubscribe_topics,
+        unsubscribe_listeners=unsubscribe_listeners,
+        mock_last_value=dict[str, Any](),
+        mock_messages=dict[str, Any](),
+        mock_discovery_all=dict[str, Any](),
+    )
+
+    with (
+        patch(f"{HA_INELS_PATH}.InelsMqtt", return_value=mqtt),
+    ):
+        yield mqtt
```

Comment:
Since you're now using the built-in broker, you could choose to just use the built-in `mqtt_mock` fixture, which will allow you to inject MQTT messages and check MQTT messages published by the iNELS integration, instead of mocking out the library. (Like you do in the test `test_mqtt_setup`).

Which route you prefer is up to you though.